packages/strapi-provider-upload-cloudinary/lib/index.js

```diff
@@ -14,7 +14,7 @@ module.exports = {
     cloudinary.config(config);
```

Comment:
the config here is the providerOptions you cannot access the options if they are set outside

Reply:
I'm not sure I understand the requested change. This is tested working as submitted and was just the least invasive way to add the options. If desired, I could possibly split the upload and delete options out to their own key at the upload config level and pull them out separately?

Reply:
I misread the example config and thought the key was on the upload config level :) 

The naming might be improved I think, `options` doesn't say much about where it is going to be used :)

make sure to also omit the key from the config you pass down to the cloudinary sdk too ;)

Reply:
I'm open to any suggestions for naming. 

As for the config object, would you be open to splitting the providerOptions object into another set of keys for providerOptions.cloudinaryConfig and perhaps something like providerOptions.cloudinaryOptions for the upload/delete options? It just makes the whole object split much more explicit rather than trying to omit a key from the unit function. 

Thanks again for taking a look at this PR. This feature will really help my particular use case where I want to use one cloudinary account for both staging and production environments without mixing the files together in root directory (options allows me to specify a folder per environment in this case to keep things organized and separate)

Reply:
I would avoid splitting for now as it will be a breaking change.

We could add another key next to the providerOptions that would be used to wrap the provider actions when we initialize it in the `createProvider` at `packages/strapi-plugin-upload/config/functions/bootstrap.js`

sth like:

```js
const createProvider = ({ provider, providerOptions, actionOptions }) => {
  const providerInstance = require(`strapi-provider-upload-${provider}`).init(providerOptions);

    return Object.assign(Object.create(baseProvider), {
      ...providerInstance,
      upload(file, options = actionOptions.upload || {}) {
        return providerInstance.upload(file, options);
      },
      delete(file, options = actionOptions.delete || {}) {
        return providerInstance.delete(file, options);
      },
    });
};
```

with this config
```js
upload: {
  provider: '',
  providerOptions: {},
  actionOptions: {
    upload: {},
    delete: {}
  }
}
```

This would make this work for any provider that wants to implement something with it. What do you think ?

Reply:
I love it, and agree on the breaking change, I'll add that and submit shortly